### PR TITLE
Switching from Travis CI to GitHub CI

### PR DESCRIPTION
**Pull request by UsamaSadiq**
**Issue:** [BOM-2111](https://openedx.atlassian.net/browse/BOM-2111)

### Description
- Added `GitHub CI` to run tests along with `Travis`.
- Updated `README.rst` to show `GitHub CI` badge status.

### Follow Up Work
- `Travis` will be removed in the follow-up PR. 
- An SRE ticket will be made to fix the coverage report issue.

### CODE REVIEW

File: .github/workflows/pypi-publish.yml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+      - name: Install pip
+        run: pip install -r requirements/pip.txt
+
+      - name: Build package
+        run: python setup.py sdist bdist_wheel
+
+      - name: Publish to PyPi
+        uses: pypa/gh-action-pypi-publish@master
+        with:
+          user: edx
```
Comment (mraarif) [inline .github/workflows/pypi-publish.yml:28]: this should be `user: __token__`

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,45 @@
+name: Python CI
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
```
Comment (jmbowman) [inline .github/workflows/ci.yml:7]: Ned has suggested that we run tests on PRs for all branches, not just those targeting master: https://edx-internal.slack.com/archives/CD89W7Z7F/p1606849364465500

File: .github/workflows/pypi-publish.yml [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+  push:
+    runs-on: ubuntu-20.04
+
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: setup python
+        uses: actions/setup-python@v2
+        with:
+          python-version: 3.5
```
Comment (jmbowman) [inline .github/workflows/pypi-publish.yml:17]: Can we run this on 3.8 instead?

File: Makefile [hunk 4]
```diff
@@ -19,6 +20,7 @@ upgrade: export CUSTOM_COMPILE_COMMAND=make upgrade
 upgrade: ## update the requirements/*.txt files with the latest packages satisfying requirements/*.in
 	pip install -q -r requirements/pip_tools.txt
 	pip-compile --upgrade -o requirements/pip_tools.txt requirements/pip_tools.in
+	pip-compile --rebuild --upgrade --allow-unsafe -o requirements/pip.txt requirements/pip.in
```
Comment (jmbowman) [inline Makefile:23]: The `--rebuild` option should be on the first pip-compile run, and you added this below the one for `pip_tools.in`.

File: requirements/base.txt [hunk 5]
```diff
@@ -4,6 +4,6 @@
 #
 #    make upgrade
 #
-django==2.2.14            # via -c requirements/constraints.txt, -r requirements/base.in
-pytz==2020.1              # via django
-sqlparse==0.3.1           # via django
+django==2.2.17            # via -c requirements/constraints.txt, -r requirements/base.in
```
Comment (jmbowman) [inline requirements/base.txt:7]: Looks like we still need a `make upgrade` job for this repo.

File: .github/workflows/pypi-publish.yml (hunk 1 above)
Comment (UsamaSadiq) [reply .github/workflows/pypi-publish.yml]: Added the `__token__` as user.

File: .github/workflows/ci.yml (hunk 2 above)
Comment (UsamaSadiq) [reply .github/workflows/ci.yml]: Added check to run on all branches.

File: .github/workflows/pypi-publish.yml (hunk 3 above)
Comment (UsamaSadiq) [reply .github/workflows/pypi-publish.yml]: Fixed to run on `python3.8`.

File: Makefile (hunk 4 above)
Comment (UsamaSadiq) [reply Makefile:22]: Made the change as per suggestion.

File: requirements/base.txt (hunk 5 above)
Comment (UsamaSadiq) [reply requirements/base.txt:7]: Ran `make upgrade` again to fetch the new versions.

**Review by jmbowman — APPROVED** [review]

**Pull request merged**